### PR TITLE
cephadm: wait longer for things to come up

### PR DESCRIPTION
**Pull request by liewegas**
- wait longer for grafana
- wait longer for mon/mgr/dashboard during bootstrap

### CODE REVIEW

**Comment by sebastian-philipp** [comment]
Hm:

```
2020-02-12T14:46:46.839 INFO:tasks.workunit.client.0.smithi181.stderr:+ sudo /tmp/tmp.70BgKXxrPt/cephadm --image ceph/daemon-base:latest-master-devel bootstrap --mon-id a --mgr-id x --mon-ip 127.0.0.1 --fsid 00000000-0000-0000-0000-0000deadbeef --config /tmp/tmp.wZcJcQehCA/tmp.0K4TXxlh5A --output-config /tmp/tmp.wZcJcQehCA/tmp.Y4NOWC2mJ8 --output-keyring /tmp/tmp.wZcJcQehCA/tmp.D2K4hcU9E5 --allow-overwrite
2020-02-12T14:46:46.912 INFO:tasks.workunit.client.0.smithi181.stderr:INFO:root:Cluster fsid: 00000000-0000-0000-0000-0000deadbeef
2020-02-12T14:46:46.913 INFO:tasks.workunit.client.0.smithi181.stderr:INFO:cephadm:Verifying IP 127.0.0.1 port 3300 ...
2020-02-12T14:46:46.913 INFO:tasks.workunit.client.0.smithi181.stderr:INFO:cephadm:Verifying IP 127.0.0.1 port 6789 ...
2020-02-12T14:46:46.913 INFO:tasks.workunit.client.0.smithi181.stderr:INFO:cephadm:Pulling latest ceph/daemon-base:latest-master-devel container...
2020-02-12T14:46:48.255 INFO:tasks.workunit.client.0.smithi181.stderr:INFO:cephadm:Extracting ceph user uid/gid from container image...
2020-02-12T14:46:49.500 INFO:tasks.workunit.client.0.smithi181.stderr:INFO:cephadm:Creating initial keys...
2020-02-12T14:46:54.126 INFO:tasks.workunit.client.0.smithi181.stderr:INFO:cephadm:Creating initial monmap...
2020-02-12T14:46:55.323 INFO:tasks.workunit.client.0.smithi181.stderr:INFO:cephadm:Creating mon...
2020-02-12T14:46:57.017 INFO:tasks.workunit.client.0.smithi181.stderr:INFO:cephadm:Waiting for mon to start...
2020-02-12T14:46:57.017 INFO:tasks.workunit.client.0.smithi181.stderr:INFO:cephadm:Waiting for mon...
2020-02-12T14:48:27.060 INFO:tasks.workunit.client.0.smithi181.stderr:INFO:cephadm:/usr/bin/ceph:timeout after 90 seconds
2020-02-12T14:48:27.068 INFO:tasks.workunit.client.0.smithi181.stderr:INFO:cephadm:Non-zero exit code -9 from /usr/bin/podman run --rm --net=host -e CONTAINER_IMAGE=ceph/daemon-base:latest-master-devel -e NODE_NAME=smithi181 -v /var/lib/ceph/00000000-0000-0000-0000-0000deadbeef/mon.a:/var/lib/ceph/mon/ceph-a:z -v /tmp/ceph-tmpc6ED5S:/etc/ceph/ceph.client.admin.keyring:z -v /tmp/ceph-tmprGWrdr:/etc/ceph/ceph.conf:z --entrypoint /usr/bin/ceph ceph/daemon-base:latest-master-devel status
2020-02-12T14:48:27.068 INFO:tasks.workunit.client.0.smithi181.stderr:INFO:cephadm:mon not available, waiting (1/10)...
2020-02-12T14:49:58.112 INFO:tasks.workunit.client.0.smithi181.stderr:INFO:cephadm:/usr/bin/ceph:timeout after 90 seconds
2020-02-12T14:49:58.124 INFO:tasks.workunit.client.0.smithi181.stderr:INFO:cephadm:Non-zero exit code -9 from /usr/bin/podman run --rm --net=host -e CONTAINER_IMAGE=ceph/daemon-base:latest-master-devel -e NODE_NAME=smithi181 -v /var/lib/ceph/00000000-0000-0000-0000-0000deadbeef/mon.a:/var/lib/ceph/mon/ceph-a:z -v /tmp/ceph-tmpc6ED5S:/etc/ceph/ceph.client.admin.keyring:z -v /tmp/ceph-tmprGWrdr:/etc/ceph/ceph.conf:z --entrypoint /usr/bin/ceph ceph/daemon-base:latest-master-devel status
2020-02-12T14:49:58.125 INFO:tasks.workunit.client.0.smithi181.stderr:INFO:cephadm:mon not available, waiting (2/10)...
2020-02-12T14:51:29.220 INFO:tasks.workunit.client.0.smithi181.stderr:INFO:cephadm:/usr/bin/ceph:timeout after 90 seconds
2020-02-12T14:51:29.228 INFO:tasks.workunit.client.0.smithi181.stderr:INFO:cephadm:Non-zero exit code -9 from /usr/bin/podman run --rm --net=host -e CONTAINER_IMAGE=ceph/daemon-base:latest-master-devel -e NODE_NAME=smithi181 -v /var/lib/ceph/00000000-0000-0000-0000-0000deadbeef/mon.a:/var/lib/ceph/mon/ceph-a:z -v /tmp/ceph-tmpc6ED5S:/etc/ceph/ceph.client.admin.keyring:z -v /tmp/ceph-tmprGWrdr:/etc/ceph/ceph.conf:z --entrypoint /usr/bin/ceph ceph/daemon-base:latest-master-devel status
2020-02-12T14:51:29.228 INFO:tasks.workunit.client.0.smithi181.stderr:INFO:cephadm:mon not available, waiting (3/10)...
2020-02-12T14:53:00.269 INFO:tasks.workunit.client.0.smithi181.stderr:INFO:cephadm:/usr/bin/ceph:timeout after 90 seconds
2020-02-12T14:53:00.276 INFO:tasks.workunit.client.0.smithi181.stderr:INFO:cephadm:Non-zero exit code -9 from /usr/bin/podman run --rm --net=host -e CONTAINER_IMAGE=ceph/daemon-base:latest-master-devel -e NODE_NAME=smithi181 -v /var/lib/ceph/00000000-0000-0000-0000-0000deadbeef/mon.a:/var/lib/ceph/mon/ceph-a:z -v /tmp/ceph-tmpc6ED5S:/etc/ceph/ceph.client.admin.keyring:z -v /tmp/ceph-tmprGWrdr:/etc/ceph/ceph.conf:z --entrypoint /usr/bin/ceph ceph/daemon-base:latest-master-devel status
2020-02-12T14:53:00.276 INFO:tasks.workunit.client.0.smithi181.stderr:INFO:cephadm:mon not available, waiting (4/10)...
2020-02-12T14:54:31.373 INFO:tasks.workunit.client.0.smithi181.stderr:INFO:cephadm:/usr/bin/ceph:timeout after 90 seconds
2020-02-12T14:54:31.384 INFO:tasks.workunit.client.0.smithi181.stderr:INFO:cephadm:Non-zero exit code -9 from /usr/bin/podman run --rm --net=host -e CONTAINER_IMAGE=ceph/daemon-base:latest-master-devel -e NODE_NAME=smithi181 -v /var/lib/ceph/00000000-0000-0000-0000-0000deadbeef/mon.a:/var/lib/ceph/mon/ceph-a:z -v /tmp/ceph-tmpc6ED5S:/etc/ceph/ceph.client.admin.keyring:z -v /tmp/ceph-tmprGWrdr:/etc/ceph/ceph.conf:z --entrypoint /usr/bin/ceph ceph/daemon-base:latest-master-devel status
2020-02-12T14:54:31.384 INFO:tasks.workunit.client.0.smithi181.stderr:INFO:cephadm:mon not available, waiting (5/10)...
2020-02-12T14:56:02.430 INFO:tasks.workunit.client.0.smithi181.stderr:INFO:cephadm:/usr/bin/ceph:timeout after 90 seconds
2020-02-12T14:56:02.440 INFO:tasks.workunit.client.0.smithi181.stderr:INFO:cephadm:Non-zero exit code -9 from /usr/bin/podman run --rm --net=host -e CONTAINER_IMAGE=ceph/daemon-base:latest-master-devel -e NODE_NAME=smithi181 -v /var/lib/ceph/00000000-0000-0000-0000-0000deadbeef/mon.a:/var/lib/ceph/mon/ceph-a:z -v /tmp/ceph-tmpc6ED5S:/etc/ceph/ceph.client.admin.keyring:z -v /tmp/ceph-tmprGWrdr:/etc/ceph/ceph.conf:z --entrypoint /usr/bin/ceph ceph/daemon-base:latest-master-devel status
2020-02-12T14:56:02.440 INFO:tasks.workunit.client.0.smithi181.stderr:INFO:cephadm:mon not available, waiting (6/10)...
2020-02-12T14:57:33.536 INFO:tasks.workunit.client.0.smithi181.stderr:INFO:cephadm:/usr/bin/ceph:timeout after 90 seconds
2020-02-12T14:57:33.544 INFO:tasks.workunit.client.0.smithi181.stderr:INFO:cephadm:Non-zero exit code -9 from /usr/bin/podman run --rm --net=host -e CONTAINER_IMAGE=ceph/daemon-base:latest-master-devel -e NODE_NAME=smithi181 -v /var/lib/ceph/00000000-0000-0000-0000-0000deadbeef/mon.a:/var/lib/ceph/mon/ceph-a:z -v /tmp/ceph-tmpc6ED5S:/etc/ceph/ceph.client.admin.keyring:z -v /tmp/ceph-tmprGWrdr:/etc/ceph/ceph.conf:z --entrypoint /usr/bin/ceph ceph/daemon-base:latest-master-devel status
2020-02-12T14:57:33.545 INFO:tasks.workunit.client.0.smithi181.stderr:INFO:cephadm:mon not available, waiting (7/10)...
2020-02-12T14:59:04.568 INFO:tasks.workunit.client.0.smithi181.stderr:INFO:cephadm:/usr/bin/ceph:timeout after 90 seconds
2020-02-12T14:59:04.576 INFO:tasks.workunit.client.0.smithi181.stderr:INFO:cephadm:Non-zero exit code -9 from /usr/bin/podman run --rm --net=host -e CONTAINER_IMAGE=ceph/daemon-base:latest-master-devel -e NODE_NAME=smithi181 -v /var/lib/ceph/00000000-0000-0000-0000-0000deadbeef/mon.a:/var/lib/ceph/mon/ceph-a:z -v /tmp/ceph-tmpc6ED5S:/etc/ceph/ceph.client.admin.keyring:z -v /tmp/ceph-tmprGWrdr:/etc/ceph/ceph.conf:z --entrypoint /usr/bin/ceph ceph/daemon-base:latest-master-devel status
2020-02-12T14:59:04.577 INFO:tasks.workunit.client.0.smithi181.stderr:INFO:cephadm:mon not available, waiting (8/10)...
2020-02-12T15:00:35.621 INFO:tasks.workunit.client.0.smithi181.stderr:INFO:cephadm:/usr/bin/ceph:timeout after 90 seconds
2020-02-12T15:00:35.629 INFO:tasks.workunit.client.0.smithi181.stderr:INFO:cephadm:Non-zero exit code -9 from /usr/bin/podman run --rm --net=host -e CONTAINER_IMAGE=ceph/daemon-base:latest-master-devel -e NODE_NAME=smithi181 -v /var/lib/ceph/00000000-0000-0000-0000-0000deadbeef/mon.a:/var/lib/ceph/mon/ceph-a:z -v /tmp/ceph-tmpc6ED5S:/etc/ceph/ceph.client.admin.keyring:z -v /tmp/ceph-tmprGWrdr:/etc/ceph/ceph.conf:z --entrypoint /usr/bin/ceph ceph/daemon-base:latest-master-devel status
2020-02-12T15:00:35.629 INFO:tasks.workunit.client.0.smithi181.stderr:INFO:cephadm:mon not available, waiting (9/10)...
2020-02-12T15:02:06.674 INFO:tasks.workunit.client.0.smithi181.stderr:INFO:cephadm:/usr/bin/ceph:timeout after 90 seconds
2020-02-12T15:02:06.681 INFO:tasks.workunit.client.0.smithi181.stderr:INFO:cephadm:Non-zero exit code -9 from /usr/bin/podman run --rm --net=host -e CONTAINER_IMAGE=ceph/daemon-base:latest-master-devel -e NODE_NAME=smithi181 -v /var/lib/ceph/00000000-0000-0000-0000-0000deadbeef/mon.a:/var/lib/ceph/mon/ceph-a:z -v /tmp/ceph-tmpc6ED5S:/etc/ceph/ceph.client.admin.keyring:z -v /tmp/ceph-tmprGWrdr:/etc/ceph/ceph.conf:z --entrypoint /usr/bin/ceph ceph/daemon-base:latest-master-devel status
2020-02-12T15:02:06.681 INFO:tasks.workunit.client.0.smithi181.stderr:INFO:cephadm:mon not available, waiting (10/10)...
2020-02-12T15:03:37.777 INFO:tasks.workunit.client.0.smithi181.stderr:INFO:cephadm:/usr/bin/ceph:timeout after 90 seconds
2020-02-12T15:03:37.789 INFO:tasks.workunit.client.0.smithi181.stderr:INFO:cephadm:Non-zero exit code -9 from /usr/bin/podman run --rm --net=host -e CONTAINER_IMAGE=ceph/daemon-base:latest-master-devel -e NODE_NAME=smithi181 -v /var/lib/ceph/00000000-0000-0000-0000-0000deadbeef/mon.a:/var/lib/ceph/mon/ceph-a:z -v /tmp/ceph-tmpc6ED5S:/etc/ceph/ceph.client.admin.keyring:z -v /tmp/ceph-tmprGWrdr:/etc/ceph/ceph.conf:z --entrypoint /usr/bin/ceph ceph/daemon-base:latest-master-devel status
2020-02-12T15:03:37.789 INFO:tasks.workunit.client.0.smithi181.stderr:ERROR: mon not available after 10 tries
```
http://qa-proxy.ceph.com/teuthology/sage-2020-02-12_14:24:10-rados-wip-sage-testing-2020-02-11-1951-distro-basic-smithi/4757698/teuthology.log

File: qa/workunits/cephadm/test_cephadm.sh [hunk 1]
```diff
@@ -120,6 +120,11 @@ cat <<EOF > $ORIG_CONFIG
 [global]
 	log to file = true
 EOF
+
+$SUDO netstat -anp
+$SUDO docker ps -a || $SUDO podman ps -a
+$SUDO journalctl -u ceph-$FSID@mon.a | cat
```
Comment (sebastian-philipp) [inline qa/workunits/cephadm/test_cephadm.sh]: which is basically equal to `cephadm logs --name mon.a | cat`.  I'd be +1 for adding this, not just temporarily!

File: qa/workunits/cephadm/test_cephadm.sh [hunk 2]
```diff
@@ -313,7 +313,8 @@ $CEPHADM enter --fsid $FSID --name mgr.x -- test -d /var/lib/ceph/mgr/ceph-x
 $CEPHADM enter --fsid $FSID --name mon.a -- pidof ceph-mon
 expect_false $CEPHADM enter --fsid $FSID --name mgr.x -- pidof ceph-mon
 $CEPHADM enter --fsid $FSID --name mgr.x -- pidof ceph-mgr
-expect_false $CEPHADM --timeout 1 enter --fsid $FSID --name mon.a -- sleep 10
+# this triggers a bug in older versions of podman, including 18.04's 1.6.2
```
Comment (mgfritch) [inline qa/workunits/cephadm/test_cephadm.sh]: The ref'd libpod issue claims fixed in 1.5.1, but we see this on 1.6.2? 

Might be nice if we could still run the tests when `podman_ver >= fixed_ver`

File: qa/workunits/cephadm/test_cephadm.sh [hunk 3]
```diff
@@ -259,7 +259,7 @@ while true; do
 	break
     fi
     TRIES=$(($TRIES + 1))
-    if [ "$TRIES" -eq 5 ]; then
+    if [ "$TRIES" -eq 30 ]; then
```
Comment (mgfritch) [inline qa/workunits/cephadm/test_cephadm.sh]: Grafana takes awhile before it's available .. I struggled here too. :+1:

File: qa/workunits/cephadm/test_cephadm.sh (hunk 2 above)
Comment (liewegas) [reply qa/workunits/cephadm/test_cephadm.sh]: Yeah, :man_shrugging: 1.6.2-1~ubuntu18.04~ppa1 is the ubunut package version, and it was definitely happening on ubuntu podman only, and commenting out that line fixed it.  Maybe it was a different/related bug that followed.

**Pull request merged**